SSL database connection should be disabled for ct-logs-discovery-provider

**Describe the bug**

SSL database connection should be disabled for ct-logs-discovery-provider in case using PgBouncer. If SSL is not disabled, connector wont start and produces error:
```
[2025-04-03 09:00:37] INFO  [/opt/czertainly/entry.sh] (process:1) Launching CT Logs Discovery Provider
2025-04-03T09:00:37.372Z  INFO  Starting CZERTAINLY-CT-Logs-Discovery-Provider  {"version": "1.0.0"}
2025-04-03T09:00:37.376Z  ERROR  Error creating schema  {"error": "failed to connect to `host=pg-bouncer-service user=czertainlyuser database=czertainlydb`: tls error (server refused TLS connection)"}
2025-04-03T09:00:37.376Z  INFO  Migrating database
2025-04-03T09:00:37.377Z  ERROR  pq: SSL is not enabled on the server
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x7087d0]
 ```

Now it possible use env:
```
ctLogsDiscoveryProvider:
  additionalEnv:
    variables:
      - name: DATABASE_SSL_MODE
        value: "disable"
```
but I think it should be automatic based on fact that PgBouncer container is deployed.

--- a/charts/ct-logs-discovery-provider/README.md
+++ b/charts/ct-logs-discovery-provider/README.md
@@ -100,38 +100,39 @@ Global values are used to define common parameters for the chart and all its sub
 
 The following values may be configured:
 
-| Parameter                                    | Default value                           | Description                                                           |
-|----------------------------------------------|-----------------------------------------|-----------------------------------------------------------------------|
-| database.type                                | `"postgresql"`                          | Type of the database, currently only `postgresql` is supported        |
-| database.host                                | `"host.docker.internal"`                | Host where is the database located                                    |
-| database.port                                | `5432`                                  | Port on which is the database listening                               |
-| database.name                                | `"czertainlydb"`                        | Database name                                                         |
-| database.schema                              | `"ctlogs"`                              | Database schema name                                                  |
-| database.username                            | `"czertainlyuser"`                      | Username to access the database                                       |
-| database.password                            | `"your-strong-password"`                | Password to access the database                                       |
-| trusted.certificates                         | `""`                                    | List of additional CA certificates that should be trusted             |
-| httpProxy                                    | `""`                                    | Proxy to be used to access external resources through http            |
-| httpsProxy                                   | `""`                                    | Proxy to be used to access external resources through https           |
-| httpProxy                                    | `""`                                    | Defines list of external resources that should not use proxy settings |
-| image.registry                               | `docker.io`                             | Docker registry name for the image                                    |
-| image.repository                             | `czertainly`                            | Docker image repository name                                          |
-| image.name                                   | `czertainly-ct-logs-discovery-provider` | Docker image name                                                     |
-| image.tag                                    | `1.0.1`                                 | Docker image tag                                                      |
-| image.digest                                 | `""`                                    | Docker image digest, will override tag if specified                   |
-| image.pullPolicy                             | `IfNotPresent`                          | Image pull policy                                                     |
-| image.pullSecrets                            | `[]`                                    | Array of secret names for image pull                                  |
-| image.command                                | `[]`                                    | Override the default command                                          |
-| image.args                                   | `[]`                                    | Override the default args                                             |
-| image.securityContext.runAsNonRoot           | `true`                                  | Run the container as non-root user                                    |
-| image.securityContext.runAsUser              | `10001`                                 | User ID for the container                                             |
-| image.securityContext.readOnlyRootFilesystem | `true`                                  | Run the container with read-only root filesystem                      |
-| image.resources                              | `{}`                                    | The resources for the container                                       |
-| podLabels                                    | `{}`                                    | Labels to be added to the pod                                         |
-| podAnnotations                               | `{}`                                    | Annotations to be added to the pod                                    |
-| podSecurityContext                           | `{}`                                    | Pod security context                                                  |
-| logging.level                                | `"INFO"`                                | Allowed values are `"INFO"`, `"DEBUG"`, `"WARN"`, `"TRACE"`           |
-| service.type                                 | `"ClusterIP"`                           | Type of the service that is exposed                                   |
-| service.port                                 | `8080`                                  | Port number of the exposed service                                    |
+| Parameter                                    | Default value                           | Description                                                                                                                                   |
+|----------------------------------------------|-----------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
+| database.type                                | `"postgresql"`                          | Type of the database, currently only `postgresql` is supported                                                                                |
+| database.host                                | `"host.docker.internal"`                | Host where is the database located                                                                                                            |
+| database.port                                | `5432`                                  | Port on which is the database listening                                                                                                       |
+| database.name                                | `"czertainlydb"`                        | Database name                                                                                                                                 |
+| database.schema                              | `"ctlogs"`                              | Database schema name                                                                                                                          |
+| database.sslMode                             | `"disable"`                             | SSL mode for the database connection, see [possible values](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-SSLMODE) |
+| database.username                            | `"czertainlyuser"`                      | Username to access the database                                                                                                               |
+| database.password                            | `"your-strong-password"`                | Password to access the database                                                                                                               |
+| trusted.certificates                         | `""`                                    | List of additional CA certificates that should be trusted                                                                                     |
+| httpProxy                                    | `""`                                    | Proxy to be used to access external resources through http                                                                                    |
+| httpsProxy                                   | `""`                                    | Proxy to be used to access external resources through https                                                                                   |
+| httpProxy                                    | `""`                                    | Defines list of external resources that should not use proxy settings                                                                         |
+| image.registry                               | `docker.io`                             | Docker registry name for the image                                                                                                            |
+| image.repository                             | `czertainly`                            | Docker image repository name                                                                                                                  |
+| image.name                                   | `czertainly-ct-logs-discovery-provider` | Docker image name                                                                                                                             |
+| image.tag                                    | `1.0.1`                                 | Docker image tag                                                                                                                              |
+| image.digest                                 | `""`                                    | Docker image digest, will override tag if specified                                                                                           |
+| image.pullPolicy                             | `IfNotPresent`                          | Image pull policy                                                                                                                             |
+| image.pullSecrets                            | `[]`                                    | Array of secret names for image pull                                                                                                          |
+| image.command                                | `[]`                                    | Override the default command                                                                                                                  |
+| image.args                                   | `[]`                                    | Override the default args                                                                                                                     |
+| image.securityContext.runAsNonRoot           | `true`                                  | Run the container as non-root user                                                                                                            |
+| image.securityContext.runAsUser              | `10001`                                 | User ID for the container                                                                                                                     |
+| image.securityContext.readOnlyRootFilesystem | `true`                                  | Run the container with read-only root filesystem                                                                                              |
+| image.resources                              | `{}`                                    | The resources for the container                                                                                                               |
+| podLabels                                    | `{}`                                    | Labels to be added to the pod                                                                                                                 |
+| podAnnotations                               | `{}`                                    | Annotations to be added to the pod                                                                                                            |
+| podSecurityContext                           | `{}`                                    | Pod security context                                                                                                                          |
+| logging.level                                | `"INFO"`                                | Allowed values are `"INFO"`, `"DEBUG"`, `"WARN"`, `"TRACE"`                                                                                   |
+| service.type                                 | `"ClusterIP"`                           | Type of the service that is exposed                                                                                                           |
+| service.port                                 | `8080`                                  | Port number of the exposed service                                                                                                            |
 
 #### Customization parameters
 

--- a/charts/ct-logs-discovery-provider/templates/hashicorp-vault-connector-deployment.yaml
+++ b/charts/ct-logs-discovery-provider/templates/hashicorp-vault-connector-deployment.yaml
@@ -68,6 +68,8 @@ spec:
                   key: database_name
             - name: DATABASE_SCHEMA
               value: {{ .Values.database.schema | quote }}
+            - name: DATABASE_SSL_MODE
+              value: {{ .Values.database.sslMode | quote }}
             - name: DATABASE_USER
               valueFrom:
                 secretKeyRef:

--- a/charts/ct-logs-discovery-provider/values.yaml
+++ b/charts/ct-logs-discovery-provider/values.yaml
@@ -69,6 +69,7 @@ database:
   port: 5432
   name: "czertainlydb"
   schema: "ctlogs"
+  sslMode: "disable"
   username: "czertainlyuser"
   password: "your-strong-password"
 

--- a/charts/hashicorp-vault-connector/README.md
+++ b/charts/hashicorp-vault-connector/README.md
@@ -100,38 +100,39 @@ Global values are used to define common parameters for the chart and all its sub
 
 The following values may be configured:
 
-| Parameter                                    | Default value                          | Description                                                           |
-|----------------------------------------------|----------------------------------------|-----------------------------------------------------------------------|
-| database.type                                | `"postgresql"`                         | Type of the database, currently only `postgresql` is supported        |
-| database.host                                | `"host.docker.internal"`               | Host where is the database located                                    |
-| database.port                                | `5432`                                 | Port on which is the database listening                               |
-| database.name                                | `"czertainlydb"`                       | Database name                                                         |
-| database.schema                              | `"hvault"`                             | Database schema name                                                  |
-| database.username                            | `"czertainlyuser"`                     | Username to access the database                                       |
-| database.password                            | `"your-strong-password"`               | Password to access the database                                       |
-| trusted.certificates                         | `""`                                   | List of additional CA certificates that should be trusted             |
-| httpProxy                                    | `""`                                   | Proxy to be used to access external resources through http            |
-| httpsProxy                                   | `""`                                   | Proxy to be used to access external resources through https           |
-| httpProxy                                    | `""`                                   | Defines list of external resources that should not use proxy settings |
-| image.registry                               | `docker.io`                            | Docker registry name for the image                                    |
-| image.repository                             | `czertainly`                           | Docker image repository name                                          |
-| image.name                                   | `czertainly-hashicorp-vault-connector` | Docker image name                                                     |
-| image.tag                                    | `1.1.2`                                | Docker image tag                                                      |
-| image.digest                                 | `""`                                   | Docker image digest, will override tag if specified                   |
-| image.pullPolicy                             | `IfNotPresent`                         | Image pull policy                                                     |
-| image.pullSecrets                            | `[]`                                   | Array of secret names for image pull                                  |
-| image.command                                | `[]`                                   | Override the default command                                          |
-| image.args                                   | `[]`                                   | Override the default args                                             |
-| image.securityContext.runAsNonRoot           | `true`                                 | Run the container as non-root user                                    |
-| image.securityContext.runAsUser              | `10001`                                | User ID for the container                                             |
-| image.securityContext.readOnlyRootFilesystem | `true`                                 | Run the container with read-only root filesystem                      |
-| image.resources                              | `{}`                                   | The resources for the container                                       |
-| podLabels                                    | `{}`                                   | Additional labels for the pod                                         |
-| podAnnotations                               | `{}`                                   | Additional annotations for the pod                                    |
-| podSecurityContext                           | `{}`                                   | Pod security context                                                  |
-| logging.level                                | `"INFO"`                               | Allowed values are `"INFO"`, `"DEBUG"`, `"WARN"`, `"TRACE"`           |
-| service.type                                 | `"ClusterIP"`                          | Type of the service that is exposed                                   |
-| service.port                                 | `8080`                                 | Port number of the exposed service                                    |
+| Parameter                                    | Default value                          | Description                                                                                                                                   |
+|----------------------------------------------|----------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
+| database.type                                | `"postgresql"`                         | Type of the database, currently only `postgresql` is supported                                                                                |
+| database.host                                | `"host.docker.internal"`               | Host where is the database located                                                                                                            |
+| database.port                                | `5432`                                 | Port on which is the database listening                                                                                                       |
+| database.name                                | `"czertainlydb"`                       | Database name                                                                                                                                 |
+| database.schema                              | `"hvault"`                             | Database schema name                                                                                                                          |
+| database.sslMode                             | `"disable"`                            | SSL mode for the database connection, see [possible values](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-SSLMODE) |
+| database.username                            | `"czertainlyuser"`                     | Username to access the database                                                                                                               |
+| database.password                            | `"your-strong-password"`               | Password to access the database                                                                                                               |
+| trusted.certificates                         | `""`                                   | List of additional CA certificates that should be trusted                                                                                     |
+| httpProxy                                    | `""`                                   | Proxy to be used to access external resources through http                                                                                    |
+| httpsProxy                                   | `""`                                   | Proxy to be used to access external resources through https                                                                                   |
+| httpProxy                                    | `""`                                   | Defines list of external resources that should not use proxy settings                                                                         |
+| image.registry                               | `docker.io`                            | Docker registry name for the image                                                                                                            |
+| image.repository                             | `czertainly`                           | Docker image repository name                                                                                                                  |
+| image.name                                   | `czertainly-hashicorp-vault-connector` | Docker image name                                                                                                                             |
+| image.tag                                    | `1.1.2`                                | Docker image tag                                                                                                                              |
+| image.digest                                 | `""`                                   | Docker image digest, will override tag if specified                                                                                           |
+| image.pullPolicy                             | `IfNotPresent`                         | Image pull policy                                                                                                                             |
+| image.pullSecrets                            | `[]`                                   | Array of secret names for image pull                                                                                                          |
+| image.command                                | `[]`                                   | Override the default command                                                                                                                  |
+| image.args                                   | `[]`                                   | Override the default args                                                                                                                     |
+| image.securityContext.runAsNonRoot           | `true`                                 | Run the container as non-root user                                                                                                            |
+| image.securityContext.runAsUser              | `10001`                                | User ID for the container                                                                                                                     |
+| image.securityContext.readOnlyRootFilesystem | `true`                                 | Run the container with read-only root filesystem                                                                                              |
+| image.resources                              | `{}`                                   | The resources for the container                                                                                                               |
+| podLabels                                    | `{}`                                   | Additional labels for the pod                                                                                                                 |
+| podAnnotations                               | `{}`                                   | Additional annotations for the pod                                                                                                            |
+| podSecurityContext                           | `{}`                                   | Pod security context                                                                                                                          |
+| logging.level                                | `"INFO"`                               | Allowed values are `"INFO"`, `"DEBUG"`, `"WARN"`, `"TRACE"`                                                                                   |
+| service.type                                 | `"ClusterIP"`                          | Type of the service that is exposed                                                                                                           |
+| service.port                                 | `8080`                                 | Port number of the exposed service                                                                                                            |
 
 #### Customization parameters
 

--- a/charts/hashicorp-vault-connector/templates/hashicorp-vault-connector-deployment.yaml
+++ b/charts/hashicorp-vault-connector/templates/hashicorp-vault-connector-deployment.yaml
@@ -68,6 +68,8 @@ spec:
                   key: database_name
             - name: DATABASE_SCHEMA
               value: {{ .Values.database.schema | quote }}
+            - name: DATABASE_SSL_MODE
+              value: {{ .Values.database.sslMode | quote }}
             - name: DATABASE_USER
               valueFrom:
                 secretKeyRef:

--- a/charts/hashicorp-vault-connector/values.yaml
+++ b/charts/hashicorp-vault-connector/values.yaml
@@ -69,6 +69,7 @@ database:
   port: 5432
   name: "czertainlydb"
   schema: "hvault"
+  sslMode: "disable"
   username: "czertainlyuser"
   password: "your-strong-password"
 

